packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -16,101 +23,165 @@ interface ScheduledAppointmentsProps {
 
 type DateType = 'pastDate' | 'today' | 'futureDate';
 
-const ScheduledAppointments: React.FC<ScheduledAppointmentsProps> = ({ visits, appointmentServiceType }) => {
-  const scheduledAppointmentsPanelsSlot = 'scheduled-appointments-panels-slot';
+const scheduledAppointmentsPanelsSlot = 'scheduled-appointments-panels-slot';
 
+const ScheduledAppointments: React.FC<ScheduledAppointmentsProps> = ({ visits, appointmentServiceType }) => {
   const { t } = useTranslation();
   const { currentAppointmentDate: date } = useAppointmentDate();
 
   const [currentTab, setCurrentTab] = useState<string>(null);
   const [dateType, setDateType] = useState<DateType>('today');
   const scheduledAppointmentPanels = useConnectedExtensions(scheduledAppointmentsPanelsSlot);
-
-  const isDateInPast = dayjs(date).isBefore(dayjs(), 'date');
-  const isDateInFuture = dayjs(date).isAfter(dayjs(), 'date');
-  const isToday = dayjs(date).isSame(dayjs(), 'date');
+  const { allowedExtensions, showExtension, hideExtension } = useAllowedExtensions();
+  const shouldShowPanel = useCallback(
+    (panel: Omit<ConnectedExtension, 'config'>) => allowedExtensions[panel.name] ?? false,
+    [allowedExtensions],
+  );
 
   useEffect(() => {
-    if (dayjs(date).isBefore(dayjs(), 'date')) {
+    const dayjsDate = dayjs(date);
+    const now = dayjs();
+    if (dayjsDate.isBefore(now, 'date')) {
       setDateType('pastDate');
-    } else if (dayjs(date).isAfter(dayjs(), 'date')) {
+    } else if (dayjsDate.isAfter(now, 'date')) {
       setDateType('futureDate');
     } else {
       setDateType('today');
     }
   }, [date]);
 
-  // set the default tab on first render
   useEffect(() => {
-    if (!currentTab) {
-      setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+    if (currentTab === null && allowedExtensions) {
```

Comment:
This is now the logic for determining which tab should be displayed if none are currently selected. Basically, the displayed tab is whatever the first extension that's allowed for the given `dateType`.

Reply:
On reflection, I think there should be some slightly more complicated logic when the `dateType` changes to ensure that the `currentTab` value still works.